src/radical/entk/appman/appmanager.py

```diff
@@ -75,7 +75,8 @@ def __init__(self,
                  rts=None,
                  rmq_cleanup=None,
                  rts_config=None,
-                 name=None):
+                 name=None,
+                 config=None):
```

Comment:
could the explicit name be used (`base_path`, `work_dir`, `session_dir`, etc.)? too many "config"s as inputs, and if to have such cumulative parameters, then all inputs should be reviewed, e.g., `rmq_settings` (which would include not just host and port, but ssl flag and vhost, and other related to RMQ), `app_settings` (which would include necessary `path`s, etc.), `rts_settings`.

Reply:
I don't mind using the explicit name, `base_path` instead of `config`. It just doesn't make sense to me that AppManager carries out all the arguments which will grow over time.

Reply:
agree about growing over time, thus would propose to have some groupings
> e.g., `rmq_settings` (which would include not just host and port, but ssl flag and vhost, and other related to RMQ), `app_settings` (which would include necessary paths, etc.), `rts_settings`.

Reply:
@lee212 , @mtitov : I agree with both of you that a growing arg list is challenging.  However, please keep in mind that RE is supposed to have a very stable API, and we should avoid breaking changes to method signatures for the time being (adding `config` is not breaking, but renaming / re-grouping would be).  

The type checking work has some impact on the API, and the upcoming changes to `name` attribute and it's interpretation also.  Maybe we should collect those issues and this config discussion into a separate milestone which allows for a (minor) iteration of the API in one release instead of spreading over multiple releases?

There was also a discussion about a more radical (ha!) overhaul of the API - but I don't think there is sufficient consensus on the need for that right now, so I would leave that off the table.

@mturilli : any opinion?

Reply:
@andre-merzky that could be done transparently - keep `**kwargs` as input and check original parameters from there, while bringing up a new approach: 
```python
def __init__(self, config_path=None, app_settings=None, <other settings>, **kwargs):

    def _get_old_inputs(**kwargs):
        hostname = kwargs.get('hostname')
        ...
```
agree to keep it as a separate ticket/PR and just to set an explicit name for a new parameter for now

Reply:
I also agree all the points made here and I don't want to break the discussion about managing API, but my reading is that we want to have `base_path` for this specific PR/ticket here instead of `config`. Because I'd like to make modifications accordingly.

Reply:
@mtitov , @andre-merzky thanks for confirming. I made a commit to use an explicit parameter name.